dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/enrollment/AbstractEnrollmentService.java

```diff
@@ -730,11 +732,12 @@ private String validateProgramForEnrollment( Program program, Enrollment enrollm
                 " is a program without registration. An enrollment cannot be created into program without registration.";
         }
 
-        if ( program.getOrganisationUnits() != null && program.getOrganisationUnits().size() > 0 )
+        IdentifiableObjectAssociations programAssociations = programService
+            .getAllProgramOrganisationUnitsAssociations( Collections.singleton( program.getUid() ) );
+        if ( programAssociations.get( program.getUid() ) != null
+            && programAssociations.get( program.getUid() ).size() > 0 )
```

Comment:
Maybe `!CollectionUtils.isEmpty(programAssociations.get( program.getUid() ) )` is more coincise

Reply:
Resolved!